vars/neoDeploy.groovy

```diff
@@ -216,11 +236,82 @@ void call(parameters = [:]) {
+                }
             }
         }
     }
 }
 
+private invalidateCache(configuration){
+    def account = configuration.neo.account
+    def host = configuration.neo.host
+    def portalLandscape = configuration.neo.portalLandscape
+
+    withCredentials([usernamePassword(
+        credentialsId: configuration.neo.oauthCredentialId,
+        passwordVariable: 'OAUTH_NEO_CLIENT_SECRET',
+        usernameVariable: 'OAUTH_NEO_CLIENT_ID')]) {
+        def bearerTokenResponse = sh(
+            script: """#!/bin/bash
+                        curl -X POST -u "${OAUTH_NEO_CLIENT_ID}:${OAUTH_NEO_CLIENT_SECRET}" \
+                            --fail \
+                            "https://oauthasservices-${account}.${host}/oauth2/api/v1/token?grant_type=client_credentials&scope=write,read"
+                    """, returnStdout: true)
+        def bearerToken = readJSON(text: bearerTokenResponse).access_token
+
+        echo "Retrieved bearer token."
+
+        def fetchXcsrfTokenResponse = sh(
+            script: """#!/bin/bash
+                        curl -i -L \
+                            -c 'cookies.jar' \
+                            -H 'X-CSRF-Token: Fetch' \
+                            -H "Authorization: Bearer ${bearerToken}" \
+                            --fail \
+                            "https://${portalLandscape}-${account}.${host}/fiori/api/v1/csrf"
+                    """, returnStdout: true)
+
+        def xcsrfToken = readProperties(text: fetchXcsrfTokenResponse)["X-CSRF-Token"]
+        def siteId = configuration.neo.siteId ?: ""
+
+        if(! siteId){
+            echo "Using the default site defined in Portal service and invalidating the cache."
+        }
+        else{
+            echo "Invalidating the cache for site with Id: ${siteId}."
+        }
+        def statusCode = sh(
+            script: """#!/bin/bash
+                        curl -X POST -L \
+                            -b 'cookies.jar'  \
+                            -H "X-CSRF-Token: ${xcsrfToken}" \
+                            -H "Authorization: Bearer ${bearerToken}" \
+                            -d "{\"siteId\":${siteId}}" \
+                            -so /dev/null \
+                            -w '%{response_code}' \
+                            "https://${portalLandscape}-${account}.${host}/fiori/v1/operations/invalidateCache"
+                    """,
+            returnStdout: true).trim()
+
+        if(! siteId && statusCode == "500") {
+            error "Invalidating the cache failed. " +
+                    "As no siteId is set, the default site defined in the portal UI is used. " +
+                    "Please verify a default site is defined in Portal service. " +
+                    "Alternatively, configure the siteId parameter for this step to invalidate the cache of that specific site."
+        }
+        else if(! statusCode == "200" || ! statusCode == "201" ){
```

Comment:
I find the code more readable if any else is on the same row as the closing bracket of the if. It's like the else belongs to the above if. But if you prefer it like this then keep it!

Reply:
done